pandas/core/dtypes/common.py

```diff
@@ -1671,7 +1671,9 @@ def _coerce_to_dtype(dtype):
     """
 
     if is_categorical_dtype(dtype):
-        dtype = CategoricalDtype()
+        categories = getattr(dtype, 'categories', None)
+        ordered = getattr(dtype, 'ordered', False)
+        dtype = CategoricalDtype(categories=categories, ordered=ordered)
```

Comment:
this is not really necessary if you allow ``CategoricalDtype(dtype)`` IOW, you can check whether the incoming ``categories`` is an instance of a ``CagetegoricalDtype`` and act accordingly.

Reply:
?

Reply:
I'm not sure I like overloading the `categories` like that.